app/models/goal.py

```diff
@@ -1,5 +1,19 @@
-from sqlalchemy.orm import Mapped, mapped_column
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+from sqlalchemy import String
 from ..db import db
 
 class Goal(db.Model):
     id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
+    title: Mapped[str] = mapped_column(String(50))
+    tasks: Mapped[list["Task"]] = relationship(back_populates="goal")
+
+    def to_dict(self):
+        return {
+            "id": self.id,
+            "title": self.title
+        }
+
+    @classmethod
+    def from_dict(cls, data):
+        return Goal(title=data["title"])
```

Comment:
While we can explicitly use `Goal` here, we can also use the `cls` keyword, like:

```suggestion
        return cls(title=data["title"])
```